WordPress/src/androidTest/java/org/wordpress/android/support/BaseTest.java

```diff
@@ -70,18 +73,41 @@ public class BaseTest {
     @Rule(order = 4)
     public WireMockRule wireMockRule;
 
-    {
-        Instrumentation instrumentation = InstrumentationRegistry.getInstrumentation();
+    public BaseTest() {
+        this(null);
+    }
 
+    /**
+     * Constructor
+     *
+     * @param wireMockFeatureFileName the wiremock feature flag file to use for this specific test.
+     */
+    public BaseTest(@Nullable final String wireMockFeatureFileName) {
+        Instrumentation instrumentation = InstrumentationRegistry.getInstrumentation();
         wireMockRule = new WireMockRule(
                 options().port(WIREMOCK_PORT)
-                         .fileSource(new AssetFileSource(instrumentation.getContext().getAssets()))
+                         .fileSource(
+                             new AssetFileSource(instrumentation.getContext().getAssets())
+                         )
+
                          .extensions(new ResponseTemplateTransformer(true, new HashMap<String, Helper>() {
                              {
                                  put("fnow", new UnlocalizedDateHelper());
                              }
                          }))
                          .notifier(new AndroidNotifier()));
+        if (wireMockFeatureFileName != null) {
+            try {
+                final String result = Utils.getStringFromInputStream(instrumentation.getContext().getClassLoader()
+                                                                                    .getResourceAsStream(
+                                                                                            wireMockFeatureFileName));
+                // This is where we can stub out
+                wireMockRule.stubFor(WireMock.get(WireMock.urlPathMatching("/wpcom/v2/mobile/feature-flags/"))
+                                             .willReturn(WireMock.aResponse().withBody(result)));
```

Comment:
We used `wireMockFeatureFileName`  as the parameter name here. I'm wondering if we can implement a more generic solution here. We added a new feature here, which is using alternative endpoint results (default results are in `/libs/mocks/src/main/assets/mocks/mappings`). I quickly checked current tests but couldn't find anywhere that we use alternative endpoint results in testing. So, you added this feature here. Let me know if my understanding is wrong.
Can we bring `/wpcom/v2/mobile/feature-flags/` as a parameter? In this way this new constructor is usable only for the `feature-flags` endpoint but we can use this for all endpoints. Wdyt?

Reply:
Yes. I didn't consider this. But I agree with your thinking. Making it generic makes sense. I'll do that. Appreciate it 🙇🏻